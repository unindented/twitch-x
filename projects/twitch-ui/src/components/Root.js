import styled from 'styled-components'

export default styled.div`
  background-color: ${props => props.theme.colors.background};
  color: ${props => props.theme.colors.foreground};
  font-family: ${props => props.theme.font.family};
  font-size: ${props => props.theme.font.sizes.base};
  line-height: ${props => props.theme.font.height};
`
