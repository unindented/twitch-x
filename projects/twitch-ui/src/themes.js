export default {
  default: {
    colors: {
      white: '#fff',
      black: '#333',
      background: '#fff',
      foreground: '#333',
      primary: '#6441a4',
      secondary: '#4b367c',
      tertiary: '#392e5c',
      quaternary: '#2c2541'
    },
    font: {
      family: 'Helvetica Neue, Helvetica, Arial, sans-serif',
      size: '16px'
    },
    layout: {
      gap: '10px',
      navCollapsedWidth: '50px',
      navExpandedWidth: '200px'
    }
  },
  tv: {
    colors: {
      white: '#fff',
      black: '#000',
      background: '#000',
      foreground: '#fff',
      primary: '#6441a4',
      secondary: '#4b367c',
      tertiary: '#392e5c',
      quaternary: '#2c2541'
    },
    font: {
      family: 'SamsungOne, sans-serif',
      size: '32px'
    },
    layout: {
      gap: '20px',
      navCollapsedWidth: '100px',
      navExpandedWidth: '400px'
    }
  }
}
