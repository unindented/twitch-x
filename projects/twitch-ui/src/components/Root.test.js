import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import themes from '../themes'
import Root from './Root'

describe('Root', () => {
  let tree

  beforeEach(() => {
    const renderer = new ShallowRenderer()
    renderer.render(<Root
      theme={themes.default}
    />)
    tree = renderer.getRenderOutput()
  })

  it('renders', () => {
    expect(tree).toMatchSnapshot()
  })
})
