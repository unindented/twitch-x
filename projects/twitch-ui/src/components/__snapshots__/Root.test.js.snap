// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders 1`] = `
<div
  className="Root-jeRvMr bIyazM"
/>
`;
